src/test/java/org/minions/devfund/reinaldo/battleship/ShipTypeEnumTest.java

```diff
@@ -0,0 +1,21 @@
+package org.minions.devfund.reinaldo.battleship;
+
+import org.junit.Test;
+
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsNull.notNullValue;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Ship type enum class.
+ */
+public class ShipTypeEnumTest {
+
+    /**
+     * Verifies if ship type exist on the enum.
+     */
+    @Test
+    public void testVerifiesTheShipTypeOnTheEnum() {
+        assertThat(ShipTypeEnum.BATTLE_CRUISER.valueOf("BATTLE_CRUISER"), is(notNullValue()));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Change this instance-reference to a static reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2209)